go/chat/boxer.go

```diff
@@ -610,9 +889,60 @@ func (b *Boxer) boxV1(messagePlaintext chat1.MessagePlaintext, key *keybase1.Cry
 	return boxed, nil
 }
 
-func (b *Boxer) boxV2(messagePlaintext chat1.MessagePlaintext, key *keybase1.CryptKey,
+func (b *Boxer) boxV2(messagePlaintext chat1.MessagePlaintext, encryptionKey *keybase1.CryptKey,
 	signingKeyPair libkb.NaclSigningKeyPair) (*chat1.MessageBoxed, error) {
-	return nil, fmt.Errorf("boxV2 not implemented")
+
+	bodyVersioned := chat1.NewBodyPlaintextWithV1(chat1.BodyPlaintextV1{
+		MessageBody: messagePlaintext.MessageBody,
+	})
+	bodyEncrypted, err := b.seal(bodyVersioned, encryptionKey)
+	if err != nil {
+		return nil, err
+	}
+
+	bodyHash := b.hashV1(bodyEncrypted.E)
```

Comment:
Since we're going through the trouble of computing the header hash with the nonce and version number thrown in, can we do the same here with the body hash?